packages/flutter_adaptive_scaffold/test/slot_layout_test.dart

```diff
@@ -0,0 +1,171 @@
+      MediaQuery slot(double width) {
+        return MediaQuery(
+          data: MediaQueryData(size: Size(width, 2000)),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: SlotLayout(
+              config: <Breakpoint, SlotLayoutConfig>{
+                TestBreakpoint0(): SlotLayout.from(
+                    key: const Key('0'), builder: (_) => const Text('Small')),
+                TestBreakpoint400(): SlotLayout.from(
+                    key: const Key('400'),
+                    builder: (_) => const Text('Medium')),
+                TestBreakpoint800(): SlotLayout.from(
+                    key: const Key('800'), builder: (_) => const Text('Large')),
+              },
+            ),
+          ),
+        );
+      }
+
+      await tester.pumpWidget(slot(300));
+      expect(find.text('Small'), findsOneWidget);
+      expect(find.text('Medium'), findsNothing);
+      expect(find.text('Large'), findsNothing);
+
+      await tester.pumpWidget(slot(500));
+      expect(find.text('Small'), findsNothing);
+      expect(find.text('Medium'), findsOneWidget);
+      expect(find.text('Large'), findsNothing);
+
+      await tester.pumpWidget(slot(1000));
+      expect(find.text('Small'), findsNothing);
+      expect(find.text('Medium'), findsNothing);
+      expect(find.text('Large'), findsOneWidget);
+    },
+  );
+
+  testWidgets(
+    'SlotLayout applies custom animations and durations correctly',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        MediaQuery(
+          data: const MediaQueryData(size: Size(500, 2000)),
+          child: Directionality(
+            textDirection: TextDirection.ltr,
+            child: SlotLayout(
+              config: <Breakpoint, SlotLayoutConfig>{
+                TestBreakpoint0(): SlotLayout.from(
+                  key: const Key('0'),
+                  builder: (_) => const SizedBox(width: 100, height: 100),
+                  inAnimation: (Widget widget, Animation<double> animation) =>
+                      ScaleTransition(
+                    scale: animation,
+                    child: widget,
+                  ),
+                  outAnimation: (Widget widget, Animation<double> animation) =>
+                      FadeTransition(
+                    opacity: animation,
+                    child: widget,
+                  ),
+                  inDuration: const Duration(seconds: 1),
+                  outDuration: const Duration(seconds: 2),
+                  inCurve: Curves.easeIn,
+                  outCurve: Curves.easeOut,
+                ),
+              },
+            ),
+          ),
+        ),
+      );
+
+      expect(find.byType(SizedBox), findsOneWidget);
+      // Verify that the animations are applied
+      final ScaleTransition scaleTransition =
+          tester.widget(find.byType(ScaleTransition));
+
+      // TODO: Fix this test
+      // final FadeTransition fadeTransition =
```

Comment:
This doesn't work because of:

```dart
if (child.key == chosenWidget?.key) {
            return (configChild.inAnimation != null)
                ? child.inAnimation!(child, animation)
                : child;
          } else {
            if (configChild.outAnimation != null) {
              hasAnimation = true;
            }
            return (configChild.outAnimation != null)
                ? child.outAnimation!(child, ReverseAnimation(animation))
                : child;
          }
```

Why is there a check on `chosenWidget?.key`? This makes the outAnimation never set.

Reply:
@diegotori do you have any idea?

Reply:
@martijn00 you'll have to ask @gspencergoog for that.

Reply:
I didn't write this, but It could be that if `chosenWidget` is null, and the `child.key` is null (a common thing), then it is matching when it wasn't intended to. Does it work if you change it to this?:

```dart
if (child.key != null && child.key == chosenWidget?.key) {
```

Reply:
Nope that doesn't work.